src/components/picker/PickerItem.tsx

```diff
@@ -30,20 +30,20 @@ const PickerItem = props => {
   const context = useContext(PickerContext);
   const {migrate} = context;
   const customRenderItem = context.renderItem || props.renderItem;
-  const itemValue = !migrate && _.isPlainObject(value) ? value?.value : value;
+  const itemValue = !migrate && typeof value === 'object' ? value?.value : value;
   const isSelected = isItemSelected(itemValue, context.value);
   const itemLabel = getItemLabel(label, value, props.getItemLabel || context.getItemLabel);
-  const selectedCounter = context.selectionLimit && context.value?.length;
+  const selectedCounter = context.selectionLimit && _.isArray(context.value) && context.value?.length;
   const accessibilityProps = {
     accessibilityState: isSelected ? {selected: true} : undefined,
     accessibilityHint: 'Double click to select this suggestion',
     ...Modifiers.extractAccessibilityProps(props)
   };
 
   const isItemDisabled = useMemo(() => {
-    return disabled || (!isSelected && context.selectionLimit && context.selectionLimit === selectedCounter);
+    return !!(disabled || (!isSelected && context.selectionLimit && context.selectionLimit === selectedCounter));
   }, [selectedCounter]);
-  
+
   useEffect(() => {
     if (_.isPlainObject(value)) {
       LogService.warn('UILib Picker.Item will stop supporting passing object as value & label (e.g {value, label}) in the next major version. Please pass separate label and value props');
```

Comment:
Why leave this log here? You have it in the Picker's hook...

Reply:
This warning related to the Picker.Item props. 
The hook is only responsible for the Picker props. 
Even though these warnings are related, because people render them separately I need to warn each component separately.